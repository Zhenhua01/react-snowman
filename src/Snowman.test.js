
import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";

describe("snapshot tests for stability", function () {
  it("matches initial", function () {
    const { container } = render(<Snowman />);
    expect(container).toMatchSnapshot();
  });
});

// update button count

describe("game playing works", function () {
  it("removes display and shows you lose message", function () {
    const { container, debug } = render(<Snowman words={["test"]} maxWrong={2}/>);
    debug(container);
    fireEvent.click(container.querySelector("button[value='a']"));

    expect(container.querySelectorAll("button[class='ltrButton']").length).toEqual(26);

    fireEvent.click(container.querySelector("button[value='b']"));

    expect(
      container.querySelector("button[class='ltrButton']")
    ).not.toBeInTheDocument();

    expect(container.querySelector(".Snowman")).toContainHTML("You lose, the word is 'test'");

  });
});

