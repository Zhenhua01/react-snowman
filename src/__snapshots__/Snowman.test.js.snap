// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for stability matches initial 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="0"
      src="0.png"
    />
    <p>
       Number wrong: 
      0
       
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
      _
    </p>
    <p>
      <button
        class="ltrButton"
        value="a"
      >
        a
      </button>
      <button
        class="ltrButton"
        value="b"
      >
        b
      </button>
      <button
        class="ltrButton"
        value="c"
      >
        c
      </button>
      <button
        class="ltrButton"
        value="d"
      >
        d
      </button>
      <button
        class="ltrButton"
        value="e"
      >
        e
      </button>
      <button
        class="ltrButton"
        value="f"
      >
        f
      </button>
      <button
        class="ltrButton"
        value="g"
      >
        g
      </button>
      <button
        class="ltrButton"
        value="h"
      >
        h
      </button>
      <button
        class="ltrButton"
        value="i"
      >
        i
      </button>
      <button
        class="ltrButton"
        value="j"
      >
        j
      </button>
      <button
        class="ltrButton"
        value="k"
      >
        k
      </button>
      <button
        class="ltrButton"
        value="l"
      >
        l
      </button>
      <button
        class="ltrButton"
        value="m"
      >
        m
      </button>
      <button
        class="ltrButton"
        value="n"
      >
        n
      </button>
      <button
        class="ltrButton"
        value="o"
      >
        o
      </button>
      <button
        class="ltrButton"
        value="p"
      >
        p
      </button>
      <button
        class="ltrButton"
        value="q"
      >
        q
      </button>
      <button
        class="ltrButton"
        value="r"
      >
        r
      </button>
      <button
        class="ltrButton"
        value="s"
      >
        s
      </button>
      <button
        class="ltrButton"
        value="t"
      >
        t
      </button>
      <button
        class="ltrButton"
        value="u"
      >
        u
      </button>
      <button
        class="ltrButton"
        value="v"
      >
        v
      </button>
      <button
        class="ltrButton"
        value="w"
      >
        w
      </button>
      <button
        class="ltrButton"
        value="x"
      >
        x
      </button>
      <button
        class="ltrButton"
        value="y"
      >
        y
      </button>
      <button
        class="ltrButton"
        value="z"
      >
        z
      </button>
    </p>
    <button
      value="restart"
    >
      Restart!
    </button>
  </div>
</div>
`;
